File: Algorithm/Code_Materials/Java/src/main/java/com/yugutou/charpter9_bitree_bisearch/level2/FindMin.java
package com.yugutou.charpter9_bitree_bisearch.level2;

/**
 * LeetCode153 旋转数字的最小数字
 */
public class FindMin {
    public static void main(String[] args) {
        int[] nums = {3, 4, 5, 1, 2};
        System.out.println(findMin(nums));
    }

    public static int findMin(int[] nums) {
        int low = 0;
        int high = nums.length - 1;
        while (low < high) {
            int pivot = low + ((high - low) >> 1);
            if (nums[pivot] < nums[high]) {
                high = pivot;
            } else {
                low = pivot + 1;
            }
        }
        return nums[low];
    }
}

File: Algorithm/Leetcode/Code_Materials/c_or_cpp/charpter9_binary/level2/find_min.cpp
#include <stdio.h>

 
int findMin(int* nums, int numsSize) {
    int low = 0;
    int high = numsSize - 1;
    while (low < high) {
        int pivot = low + (high - low) / 2;
        if (nums[pivot] < nums[high]) {
            high = pivot;
        } else {
            low = pivot + 1;
        }
    }
    return nums[low];
}

int main()
{
    int array[] = {5,6,7,1, 2, 3, 4};
    int index = findMin(array, 6);
    if (index == -1)
    {
        printf("Target not found in the array.\n");
    }
    else
    {
        printf("Target found at index %d.\n", index);
    }
    return 0;
}